### DISCUSSION TITLE
Symfony composer dependencies

### DISCUSSION THREAD

**Discussion by amigian74** [symfony/symfony#43054] [Q&A]
Hi,

I'm struggeling with the composer depencies of Symfony. Unfortunately lots of the Symfony "internal" dependenies are very vague which leads to some packages upgrading to higher versions then expected.

An example;

I've included the symfony/security-bundle bound to version 2.5* in my project. This bundle includes (right now) symfony/security-core with the version 5.3.*. Why??

Right now this behavior leads to millions of deprecation warnings which is a pain in ...

So I must include all these depentend packages myself with the right version.

Is there a better way or should'nt the version bound through the dependency tree? Maybe this is a composer problem.

Thx in advance

**Comment by mbabker** [comment]
> I've included the symfony/security-bundle bound to version 2.5* in my project.

You're positive this is 2.5 and not 5.2?

Either way, the components all use semantic versioning to declare the supported versions.  They don't all lock themselves to the same Symfony version, that would cause a lot of pain for everyone relying on any Symfony package.  So, by the B/C promise, SecurityBundle 5.2 should be able to safely use any of the Security component releases from the 5.2, 5.3, or 5.4 branches.  Sure, there might be a lot of deprecation notices, but the combination of packages you have installed should be functional.

If you need all of the Symfony components locked to the same version, you can either use Symfony Flex to help manage this or explicitly declare in your `composer.json` file the required packages and versions.
